e3sm_diags/parameter/core_parameter.py

```diff
@@ -213,3 +219,48 @@ def check_values(self):
         ):
             msg = "You need to define both the 'test_start_yr' and 'test_end_yr' parameter."
             raise RuntimeError(msg)
+
+    def _run_diag(self) -> List[Any]:
+        """Run the corresponding diagnostics for a set of parameters.
+
+        Additional CoreParameter (or CoreParameter sub-class) objects are derived
+        from the CoreParameter `sets` attribute, hence this function returns a
+        list of CoreParameter objects.
+
+        This method loops over the parameter's diagnostic sets and attempts to
+        import and call the related `run_diags()` function.
+
+        Returns
+        -------
+        List[Any]
+            The list of CoreParameter objects with results from the diagnostic run.
+            NOTE: `Self` type is not yet supported by mypy.
+        """
+        results = []
+
+        for set_name in self.sets:
+            self.current_set = set_name
+            # FIXME: This is a shortcut to importing `run_diag`, but can break
+            # easily because the module driver must match the module name.
```

Comment:
```suggestion
            # FIXME: This is a shortcut to importing `run_diag`, but can break
            # easily because the module driver include the set name (static reference).
```